Source/Provers/SMTLib/SMTLibProcessTheoremProver.cs

```diff
@@ -1522,7 +1522,7 @@ protected int ParseRCount(SExpr resp)
       {
         // If anything goes wrong with parsing the response from the solver,
         // it's better to be able to continue, even with uninformative data.
-        currentErrorHandler.OnProverWarning($"Failed to parse resource count from solver. Got: {resp}");
+        currentErrorHandler?.OnProverWarning($"Failed to parse resource count from solver. Got: {resp}");
```

Comment:
I still have a bad feeling about not stacking warnings before  an output handler is available. A nullpointerexception here would have meant that we are dismissing this warning. This is not desirable. All warnings should make their way to the user. That's probably why they were "flushed", before, when we are sure that there is a currentErrorHandler available. Could you perhaps use an async queue for that purpose?